frontend/models/product-list/server.ts

```diff
@@ -139,6 +145,23 @@ export function getProductListType(
    }
 }
 
+async function filterDevicesWithNoProducts(
+   productListChildren: ProductListChild[],
+   isPartsList: boolean
+) {
+   if (!isPartsList) {
+      return productListChildren;
+   }
+   const childDevices = productListChildren.reduce(
+      (acc, child) => (child.deviceTitle ? [...acc, child.deviceTitle] : acc),
```

Comment:
This is effectively an `O(n^2)` implementation.

How about `children.map(child => child.deviceTitle).filter(child => child.deviceTitle)`

Reply:
Good eye! This motivated me to do some learning too: https://prateeksurana.me/blog/why-using-object-spread-with-reduce-bad-idea/

Reply:
https://github.com/iFixit/react-commerce/pull/1252/commits/b9e1765f1ad39c7e3a53eeedb8e3dde1163a11b9